### ISSUE TITLE
Investigate results of 02_06

### ISSUE THREAD

**Issue by rviscomi** [HTTPArchive/almanac.httparchive.org#187]
In the [results for 02_06](https://docs.google.com/spreadsheets/d/1uFlkuSRetjBNEhGKWpkrXo4eEIsgYelxY-qR9Pd7QpM/edit#gid=154353360) "% of pages per color format" @argyleink and @una pointed out a possible bug:

> is this right? HSL less popular than HSLA?

> woh, why would a mobile stylesheet have a random HSL value where the desktop one didnt!?

client | freq_hsl | freq_hsla | total | pct_hsl | pct_hsla
-- | -- | -- | -- | -- | --
desktop | 32456 | 824154 | 4371973 | 0.74 | 18.85
mobile | 455149 | 999030 | 5297442 | 8.59 | 18.86

So there are two questions: why is HSL less popular than HSLA? why is desktop HSL so much smaller than desktop HSLA?

Here is the [underlying query](https://github.com/HTTPArchive/almanac.httparchive.org/blob/master/sql/2019/02_CSS/02_06.sql).

**Comment by rviscomi** [comment]
Wrote a hacky query to count the number of stylesheets (not pages) that contain either `hsl(` or `hsla(`

```sql
SELECT
  client,
  COUNTIF(REGEXP_CONTAINS(body, r'hsl\(')) AS hsl,
  COUNTIF(REGEXP_CONTAINS(body, r'hsla\(')) AS hsla
FROM
  `httparchive.almanac.summary_response_bodies`
WHERE
  type = 'css'
GROUP BY
  client
```

client | hsl | hsla
-- | -- | --
mobile | 507665 | 1371314
desktop | 43319 | 1074992

It aligns with the proportions of the parsed CSS results, so I'm more confident that the results are correct.

**Comment by argyleink** [comment]
🤷‍♂

**Comment by una** [comment]
Further poll: https://twitter.com/Una/status/1181303759231684608?s=20

**Comment by rviscomi** [comment]
Dug into this more. It seems that the embedded YouTube video player stylesheets are biasing the results.

```sql
#standardSQL
SELECT
  client,
  url,
  COUNTIF(REGEXP_CONTAINS(body, r'hsl\(')) AS hsl,
  COUNTIF(REGEXP_CONTAINS(body, r'hsla\(')) AS hsla
FROM
  `httparchive.almanac.summary_response_bodies`
WHERE
  type = 'css'
GROUP BY
  client,
  url
ORDER BY
  hsl DESC
LIMIT
  100
```

For example https://www.youtube.com/yts/cssbin/www-player-sprite-mode-vflY8RB4O.css (and many variants of this URL with different VFL keys) make up hundreds of thousands of hits for HSL on mobile.

Compared to sorting by top HSLA stylesheets:

```sql
#standardSQL
SELECT
  client,
  url,
  COUNTIF(REGEXP_CONTAINS(body, r'hsl\(')) AS hsl,
  COUNTIF(REGEXP_CONTAINS(body, r'hsla\(')) AS hsla
FROM
  `httparchive.almanac.summary_response_bodies`
WHERE
  type = 'css'
GROUP BY
  client,
  url
ORDER BY
  hsla DESC
LIMIT
  100
```

Top URLs that use HSLA like https://c0.wp.com/c/5.2.2/wp-includes/css/dist/block-library/style.min.css appear to make exclusive use of HSLA, not including any HSL styles, and are in the tens of thousands.

In conclusion, hugely popular 3P content like the YT player CSS favor HSL and are only served to mobile. I wonder if this is a browser compatibility optimization? In any case, I'll close this out.

**Comment by argyleink** [comment]
interesting!
